Kitodo/src/test/java/org/kitodo/production/process/NewspaperProcessesGeneratorIT.java

```diff
@@ -0,0 +1,118 @@
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.process;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.apache.commons.lang.SystemUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.kitodo.ExecutionPermission;
+import org.kitodo.FileLoader;
+import org.kitodo.MockDatabase;
+import org.kitodo.NewspaperCourse;
+import org.kitodo.SecurityTestUtils;
+import org.kitodo.TreeDeleter;
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.enums.KitodoConfigFile;
+import org.kitodo.config.enums.ParameterCore;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.production.model.bibliography.course.Course;
+import org.kitodo.production.model.bibliography.course.Granularity;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.ProcessService;
+import org.kitodo.production.services.file.FileService;
+
+public class NewspaperProcessesGeneratorIT {
+    private static final FileService fileService = new FileService();
+    private static final ProcessService processService = ServiceManager.getProcessService();
+
+    private static final String firstProcess = "First process";
+    private static final File script = new File(ConfigCore.getParameter(ParameterCore.SCRIPT_CREATE_DIR_META));
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    /**
+     * The test environment is being set up.
+     *
+     * @throws Exception
+     *             if that does not work
+     */
+    @BeforeClass
+    public static void setUp() throws Exception {
+        if (!SystemUtils.IS_OS_WINDOWS) {
+            ExecutionPermission.setExecutePermission(script);
+        }
+        FileLoader.createConfigProjectsFileForCalendarHierarchyTests();
+        MockDatabase.startNode();
+        MockDatabase.insertProcessesFull();
+        MockDatabase.insertProcessForCalendarHierarchyTests();
+        MockDatabase.setUpAwaitility();
+        fileService.createDirectory(URI.create(""), "1");
+        SecurityTestUtils.addUserDataToSecurityContext(ServiceManager.getUserService().getById(1), 1);
+        while (Objects.isNull(processService.findByTitle(firstProcess))) {
+            Thread.sleep(256);
+        }
+    }
+
+    /**
+     * The test environment is cleaned up and the database is closed.
+     */
+    @AfterClass
+    public static void tearDown() throws Exception {
+        MockDatabase.stopNode();
+        MockDatabase.cleanDatabase();
+        KitodoConfigFile.PROJECT_CONFIGURATION.getFile().delete();
```

Comment:
why are you deleting the file?

Reply:
`ProjectServiceTest.testProjectForCompletness()` relies on the fact that this file does not exist initially. It also creates and deletes this file. In order to integrate with the other tests, it therefore seems necessary that this file is generated during during `setUp()`and deleted during `tearDown()`.